openedx-unsupported/ecommerce
# ENT-803: Added functionality to always create enrollment codes with course creation.

@brittneyexline , @zubair-arbi , @georgebabey  -- FYI
__Description:__
This PR adds the functionality to create enrollment code be default along with course creation.

**JIRA Ticket:** [ENT-803](https://openedx.atlassian.net/browse/ENT-803)

**Testing Instructions:**

1. Go to studio (https://studio-business.sandbox.edx.org/home/) and create a new course, use this course to perform the following operations.
2. Make sure `Enable enrollment codes` at https://ecommerce-business.sandbox.edx.org/admin/core/siteconfiguration/1/change/ is unchecked.
3. Make sure `create_enrollment_codes` switch at https://ecommerce-business.sandbox.edx.org/admin/waffle/switch/3/change/ is not active.
4. Go to course creation page on ecommerce at https://ecommerce-business.sandbox.edx.org/courses/new/ and create a new course with professional mode.
5. Go to Product admin list view at https://ecommerce-business.sandbox.edx.org/admin/catalogue/product/ and make sure there is a product with the above course id and `Enrollment Code` product class.
6. Repeat 1-4 with an audit course and make sure product with the above course id and `Enrollment Code` product class is not created.
